src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs

```diff
@@ -1020,7 +1025,7 @@ private Type GetProviderSpecificFieldTypeInternal(_SqlMetaData metaData)
 
                 if (metaData.type == SqlDbType.Udt)
                 {
-                    throw UdtNotSupportedException();
+                    providerSpecificFieldType = MetaType.MetaMaxVarBinary.SqlType;
```

Comment:
Why did you change this? `GetProviderSpecificFieldTypeInternal` is called by `GetProviderSpecificFieldType`
In `GetProviderSpecificFieldType` you are already throwing an exception on Udt.